### PR TITLE
docs: add migrations for arguments

### PR DESCRIPTION
**Pull request by boneskull**
This depends on #15938.

This list is certainly incomplete and just downright wrong, so I'm hoping others can help.

When the `--help` text says "iOS only" or "Android only", I am not sure which drivers those entail.  

(If it's multiple, it makes me wonder if a "base iOS" or "base Android" driver might make sense)

### CODE REVIEW

File: docs/en/advanced-concepts/migrating-to-appium-2.0.md [hunk 1]
```diff
@@ -163,7 +163,30 @@ TODO
 
 #### :tada: Driver and Plugin CLI args
 
-TODO
+#### CLI Argument Migrations
+
+Many driver-specific arguments have changed. These arguments are now only available upon installation of the appropriate driver:
+
+| Old Argument                 | New Argument                                                                                       | Driver(s)                      |
```
Comment (boneskull) [inline docs/en/advanced-concepts/migrating-to-appium-2.0.md:170]: should sort this table when it's complete

**Comment by jlipps** [comment]
@boneskull I think probably 80-85% of the args which were in 1.x but which are not in 2.x have not been moved to driver args. Instead they were deprecated and have been removed. I think it's probably better to find the PRs which added driver args to the different drivers, and mention those in terms of the migration. All the others should be listed as no longer available (instead the way to set them on the command line if desired is to make them part of `--default-capabilities`).

**Comment by jlipps** [comment]
https://github.com/appium/appium-uiautomator2-driver/pull/459/files (this one is still open, so probably is superseded by your work now?)

**Comment by jlipps** [comment]
for xcuitest: https://github.com/appium/appium-xcuitest-driver/pull/1331

**Comment by boneskull** [comment]
got it

**Comment by jlipps** [comment]
> (If it's multiple, it makes me wonder if a "base iOS" or "base Android" driver might make sense)

The current situation is that both the XCUITest driver and the UiAutomator2 driver extend other drivers (IOSDriver and AndroidDriver), which are no longer used directly themselves. It has been proposed that those projects should be removed and instead the shared code extracted into IOSBaseDriver and AndroidBaseDriver. It's a good idea and would form the basis of the (also planned) iOS and Android monorepos, which we decided we didn't need to address concretely until after Appium 2.0 was out.

**Comment by mykola-mokhnach** [comment]
> The current situation is that both the XCUITest driver and the UiAutomator2 driver extend other drivers (IOSDriver and AndroidDriver)

Not really. XCUITest driver does not depend on any legacy stuff anymore.

File: docs/en/advanced-concepts/migrating-to-appium-2.0.md [hunk 2]
```diff
@@ -163,7 +163,30 @@ TODO
 
 #### :tada: Driver and Plugin CLI args
 
-TODO
+#### CLI Argument Migrations
```
Comment (mykola-mokhnach) [inline docs/en/advanced-concepts/migrating-to-appium-2.0.md:166]: I'd say all these almost all args should be removed from v2 as they are:
- driver-specific
- could be replaced with default capabilities
It only makes sense to keep these, that a system admin would potentially like to keep, in order to override user-provided capabilities on the server side, for which we could have potentially have another common argument. Something like `enforceCapabilities`. Capabilities provided there would have priority over all other values included these provided by users

**Comment by jlipps** [comment]
> Not really. XCUITest driver does not depend on any legacy stuff anymore.

Oh cool. I missed that it didn't depend on `appium-ios-driver` anymore. Good stuff!

File: docs/en/advanced-concepts/migrating-to-appium-2.0.md [hunk 3]
```diff
@@ -163,7 +163,30 @@ TODO
...
+| `--robot-address`            | `--driver-xcuitest-robot-address`                                                                  | `xcuitest`                     |
+| `--robot-port`               | `--driver-xcuitest-robot-port`                                                                     | `xcuitest`                     |
+| `--chromedriver-executable`  | `--driver-uiautomator2-chromedriver-executable` _or_ `--driver-selendroid-chromedriver-executable` | `uiautomator2` or `selendroid` |
+| `--isolate-sim-device`       | `--driver-xcuitest-sim-device`                                                                     | `xcuitest`                     |
+| `--suppress-adb-kill-server` | `--driver-uiautomator2-adb-kill-server`                                                            | `uiautomator2`                 |
+| `--webkit-debug-proxy-port`  | `--driver-xcuitest-webkit-debug-proxy-port`                                                        | `xcuitest`                     |
+| `--webdriveragent-port`      | `--driver-xcuitest-webdriveragent-port`                                                            | `xcuitest`                     |
```
Comment (jlipps) [inline docs/en/advanced-concepts/migrating-to-appium-2.0.md:189]: keep

File: docs/en/advanced-concepts/migrating-to-appium-2.0.md [hunk 4]
```diff
@@ -163,7 +163,30 @@ TODO
...
+| `--instruments`              | `--driver-xcuitest-instruments`                                                                    | `xcuitest`                     |
+| `--robot-address`            | `--driver-xcuitest-robot-address`                                                                  | `xcuitest`                     |
+| `--robot-port`               | `--driver-xcuitest-robot-port`                                                                     | `xcuitest`                     |
+| `--chromedriver-executable`  | `--driver-uiautomator2-chromedriver-executable` _or_ `--driver-selendroid-chromedriver-executable` | `uiautomator2` or `selendroid` |
+| `--isolate-sim-device`       | `--driver-xcuitest-sim-device`                                                                     | `xcuitest`                     |
+| `--suppress-adb-kill-server` | `--driver-uiautomator2-adb-kill-server`                                                            | `uiautomator2`                 |
+| `--webkit-debug-proxy-port`  | `--driver-xcuitest-webkit-debug-proxy-port`                                                        | `xcuitest`                     |
```
Comment (jlipps) [inline docs/en/advanced-concepts/migrating-to-appium-2.0.md:188]: keep

File: docs/en/advanced-concepts/migrating-to-appium-2.0.md [hunk 5]
```diff
@@ -163,7 +163,30 @@ TODO
...
+| `--tracetemplate`            | `--driver-xcuitest-tracetemplate`                                                                  | `xcuitest`                     |
+| `--instruments`              | `--driver-xcuitest-instruments`                                                                    | `xcuitest`                     |
+| `--robot-address`            | `--driver-xcuitest-robot-address`                                                                  | `xcuitest`                     |
+| `--robot-port`               | `--driver-xcuitest-robot-port`                                                                     | `xcuitest`                     |
+| `--chromedriver-executable`  | `--driver-uiautomator2-chromedriver-executable` _or_ `--driver-selendroid-chromedriver-executable` | `uiautomator2` or `selendroid` |
+| `--isolate-sim-device`       | `--driver-xcuitest-sim-device`                                                                     | `xcuitest`                     |
+| `--suppress-adb-kill-server` | `--driver-uiautomator2-adb-kill-server`                                                            | `uiautomator2`                 |
```
Comment (jlipps) [inline docs/en/advanced-concepts/migrating-to-appium-2.0.md:187]: keep

File: docs/en/advanced-concepts/migrating-to-appium-2.0.md [hunk 6]
```diff
@@ -163,7 +163,30 @@ TODO
...
+| `--force-iphone`             | `--driver-xcuitest-force-iphone`                                                                   | `xcuitest`                     |
+| `--force-ipad`               | `--driver-xcuitest-force-ipad`                                                                     | `xcuitest`                     |
+| `--tracetemplate`            | `--driver-xcuitest-tracetemplate`                                                                  | `xcuitest`                     |
+| `--instruments`              | `--driver-xcuitest-instruments`                                                                    | `xcuitest`                     |
+| `--robot-address`            | `--driver-xcuitest-robot-address`                                                                  | `xcuitest`                     |
+| `--robot-port`               | `--driver-xcuitest-robot-port`                                                                     | `xcuitest`                     |
+| `--chromedriver-executable`  | `--driver-uiautomator2-chromedriver-executable` _or_ `--driver-selendroid-chromedriver-executable` | `uiautomator2` or `selendroid` |
```
Comment (jlipps) [inline docs/en/advanced-concepts/migrating-to-appium-2.0.md:185]: keep (only `uiautomator2`, selendroid is gone)

File: docs/en/advanced-concepts/migrating-to-appium-2.0.md [hunk 7]
```diff
@@ -163,7 +163,30 @@ TODO
...
-TODO
+#### CLI Argument Migrations
+
+Many driver-specific arguments have changed. These arguments are now only available upon installation of the appropriate driver:
+
+| Old Argument                 | New Argument                                                                                       | Driver(s)                      |
+| ---------------------------- | -------------------------------------------------------------------------------------------------- | ------------------------------ |
+| `--reboot`                   | `--driver-uiautomator2-reboot`                                                                     | `uiautomator2`                 |
+| `--ipa`                      | `--driver-xcuitest-ipa`                                                                            | `xcuitest`                     |
+| `--bootstrap-port`           | `--driver-uiautomator2-bootstrap-port`                                                             | `uiautomator2`                 |
```
Comment (jlipps) [inline docs/en/advanced-concepts/migrating-to-appium-2.0.md:174]: keep

**Review by jlipps — CHANGES_REQUESTED** [review]
i think we should keep only the args i commented 'keep' on. the others are all better accessed as driver specific capabilities.

**Pull request closed without merging**